gobblin-core/src/main/java/gobblin/writer/AvroHdfsTimePartitionedWithRecordCountsWriter.java

```diff
@@ -48,21 +53,21 @@ public AvroHdfsTimePartitionedWithRecordCountsWriter(Destination destination, St
   public void close() throws IOException {
     super.close();
 
-    String fileName = WriterUtils.getWriterFileName(this.properties, this.numBranches, this.branch, this.writerId,
-        this.writerOutputFormat.getExtension());
+    // Rewrite property writer.final.output.path due to renaming output files.
+    this.properties.removeProp(ConfigurationKeys.WRITER_FINAL_OUTPUT_PATH);
 
-    Path writerOutputDir = new Path(this.properties.getProp(ForkOperatorUtils
-        .getPropertyNameForBranch(ConfigurationKeys.WRITER_OUTPUT_DIR, this.numBranches, this.branch)));
+    for (Entry<Path, FsDataWriter<GenericRecord>> entry : this.pathToWriterMap.entrySet()) {
 
-    for (Entry<Path, DataWriter<GenericRecord>> entry : this.pathToWriterMap.entrySet()) {
-
-      String filePathOld = new Path(entry.getKey(), fileName).toString();
+      String filePathOld = entry.getValue().getOutputFilePath();
 
       String filePathNew = filePathOld.substring(0, filePathOld.lastIndexOf(".")) + "."
```

Comment:
This only return file names, but I also need the path (e.g., given `/a/b/c.txt` it returns `c` but I need `a/b/c`